server/src/main/java/org/opensearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -774,7 +780,52 @@ public synchronized void handleRecoveryFailure(ShardRouting shardRouting, boolea
 
     public void handleRecoveryDone(ReplicationState state, ShardRouting shardRouting, long primaryTerm) {
         RecoveryState RecState = (RecoveryState) state;
-        shardStateAction.shardStarted(shardRouting, primaryTerm, "after " + RecState.getRecoverySource(), SHARD_STATE_ACTION_LISTENER);
+        AllocatedIndex<? extends Shard> indexService = indicesService.indexService(shardRouting.shardId().getIndex());
+        IndexShard indexShard = (IndexShard) indexService.getShardOrNull(shardRouting.id());
+        // For Segment Replication enabled indices, we want replica shards to start a replication event to fetch latest segments before it
+        // is marked as Started.
+        if (indexShard.indexSettings().isSegRepEnabled()
+            && shardRouting.primary() == false
+            && ShardRoutingState.RELOCATING != shardRouting.state()) {
```

Comment:
Should this condition be `ShardRoutingState.STARTED == shardRouting.state()` ? Existing condition applies for UNASSIGNED and INITIALIED shards, is that correct ?